external-crates/move/crates/move-abstract-interpreter/src/control_flow_graph.rs

```diff
@@ -3,70 +3,98 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! This module defines the control-flow graph uses for bytecode verification.
-use move_binary_format::file_format::{Bytecode, CodeOffset, VariantJumpTable};
 use std::collections::{btree_map::Entry, BTreeMap, BTreeSet};
 
-// BTree/Hash agnostic type wrappers
-type Map<K, V> = BTreeMap<K, V>;
-type Set<V> = BTreeSet<V>;
-
-pub type BlockId = CodeOffset;
-
 /// A trait that specifies the basic requirements for a CFG
 pub trait ControlFlowGraph {
+    type BlockId: Copy + Ord;
+    type InstructionIndex: Copy + Ord;
+    type Instruction;
+    type Instructions: ?Sized;
```

Comment:
A very small nit on design here: we can use the other traits, below, as requirements for these types. For example, `type Instruction` could become `type Instr: Instruction` and any instantiation will require the trait. This may also allow us to provide some default implementations if we want (though plenty of arguments against those could be made).